### PR TITLE
PXB-2729 PXB compilation fails if GTest packages are installed

### PR DESCRIPTION
**Pull request by rahulmalik87**
https://jira.percona.com/browse/PXB-2729

Fixed the compilation error and xbcloud-t.cc test case

### CODE REVIEW

File: storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc [hunk 1]
```diff
@@ -226,14 +229,14 @@ TEST(s3v4_signer, storageClass) {
...
-               "891034a3bd13729689a54d363380ad1849b26bf2b9e461d4c2bdeeca32e0c1c"
-               "e");
+  ASSERT_STREQ(
+      req.headers().at("Authorization").c_str(),
+      "AWS4-HMAC-SHA256 "
+      "Credential=access_key/20190422/example-region/s3/aws4_request, "
+      "SignedHeaders=content-length;content-type;host;x-amz-content-"
+      "sha256;x-amz-date;x-amz-security-token;x-amz-storage-class, "
+      "Signature="
+      "fe6c888f22fb23a7a3fe6f663013b5df3cc761c3777ed4368b325114c885320f");
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc:241]: how did you get new signature?

File: storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc [hunk 2]
```diff
@@ -77,7 +78,8 @@ TEST(s3_client, basicDNSv4) {
 
 TEST(s3_client, basicPATHv4) {
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc:80]: Should we add similar test for azure APIs?

File: storage/innobase/xtrabackup/src/xbcloud/CMakeLists.txt [hunk 3]
```diff
@@ -71,7 +71,7 @@ TARGET_LINK_LIBRARIES(xbcloud
   crc
   )
 
-IF (WITH_DEBUG)
+IF (WITH_DEBUG AND CMAKE_VERSION VERSION_GREATER 3.22.1)
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/xbcloud/CMakeLists.txt]: any limitation with cmake < 3.22.1?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/src/xbcloud/CMakeLists.txt (hunk 3 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/xbcloud/CMakeLists.txt]: yes. GTest::gmock was introduce after 3.22.1 . 

https://cmake.org/cmake/help/v3.23/module/FindGTest.html.

File: storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc (hunk 1 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc:241]: the test was failing and copied the new signature.

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]
@rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,

We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.

Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.

**Comment by rahulmalik87** [comment]
> @rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,
> 
> We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.
> 
> Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.

Do you mean adding new CMAKE flag or using existing CMAKE that mysql uses . There is compilation error in using existing CMAKE with_unit_test flag for PXB

**Comment by altmannmarcelo** [comment]
> > @rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,
> > We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.
> > Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.
> 
> Do you mean adding new CMAKE flag or using existing CMAKE that mysql uses . There is compilation error in using existing CMAKE with_unit_test flag for PXB

@rahulmalik87 I mean using the existing one. On this issue, we found out that there is no way to compile a Debug version but disable Gunit tests if you have Gunit library installed. We should give the ability to disable its compilation.

**Comment by rahulmalik87** [comment]
> > > @rahulmalik87 I think we should not enforce the tests on WITH_DEBUG builds,
> > > We should instead change this flag with WITH_UNIT_TEST and allow it to be turned on/off at will.
> > > Maybe default to OFF on RelWithDebInfo and default to ON on Debug, but with the option to flip it.
> > 
> > 
> > Do you mean adding new CMAKE flag or using existing CMAKE that mysql uses . There is compilation error in using existing CMAKE with_unit_test flag for PXB
> 
> @rahulmalik87 I mean using the existing one. On this issue, we found out that there is no way to compile a Debug version but disable Gunit tests if you have Gunit library installed. We should give the ability to disable its compilation.

It only compiles with WITH_UNIT_TESTS

File: storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc (hunk 2 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/xbcloud/xbcloud-t.cc:80]: added

**Review by satya-bodapati — APPROVED** [review]
plz squash before merge

**Review by altmannmarcelo — APPROVED** [review]
LGTM

**Pull request merged**